### DISCUSSION TITLE
Change download location QBittorrent on Proxmox

### DISCUSSION THREAD

**Discussion by Citroene** [tteck/Proxmox#1416] [Help Needed] [answered]
Hi,

I may have a stupid question but...

I installed Qbittorrent on proxmox.
I would like to change the download location to my nas, how can I change this?

**Comment by oOStroudyOo** [comment]
Have you mounted the NAS to that container?
Is your NAS folders public or do you need credentials to access them?

For me I did the following but this could be different for yourself, depending on what your set up is:

I had to install a few prerequisites for my NAS to work (May not be required for yours, I use a WD Cloud): `sudo apt -y install software-properties-common && sudo apt-get -y install nfs-common && sudo apt-get -y install cifs-utils`

Then I created a directory to store my credentials for the access to the NAS (Not required if your NAS has public access to internal network): `mkdir .credentials`

Then I created a file that had the credentials on it (Not required if your NAS has public access to internal network): 
`nano /root/.credentials/my-credentials`

In the file the format was like this:
`username=your-username-here`
`password=your-password-here`
`domain=your-nas-IP-here`

I added permissions to the credentials file: `chmod 600 /root/.credentials/my-credentials`

Then I added a directory which I would use to access the NAS: `mkdir /media/nas`

Then I needed to edit the fstab file of the container: `nano /etc/fstab`

This is what I wrote insite the fstab file (Change the start part to whatever level of your NAS that you want): `//xxx.xxx.xxx.xxx/your/nas/directory/here /media/nas cifs uid=0,credentials=/root/.credentials/my-credentials,iocharset=utf8,noperm 0 0` 

Then mount the NAS: `mount -a`

Now when you change directory to the one you created earlier (`cd /media/wdnas`), you should be able to access the folders on your NAS.

Then load up the qBittorrent UI and sign in.
Go to settings.
Downloads tab.
Then you should be able to selelct //media/nas as a Default Save Path.

I'm sure there are many other ways of doing it, maybe you could just change the Default Save Path to //IP-Address/Downloads-Folder

**Comment by oOStroudyOo** [comment, marked as answer]
Or if you already have your NAS drives set as SMB/CIFS storage on your Datacentre, you can add them as mount points via your LXC config file:

- Click on your datacentre and go to storage.
- Make note of your Path/Target of your NAS storage.
- Click on your PVE node and open your console.
- Access the LXC config file, changing the number to your relevant container ID: `nano /etc/pve/lxc/100.conf`
- Add a new line and type in: `mp0: /the/NAS/storage/path,mp=/the/directory/to/access/the/storage/in/the/container`

So mine for example with 2 NAS storages:

mp0: /mnt/pve/NAS,mp=/media/nas4tb
mp1: /mnt/pve/NAS-6TB,mp=/media/nas6tb

**/mnt/pve/NAS** and **/mnt/pve/NAS-6TB** is the path from my Promox datacentre to my NAS drives.
**/media/nas4tb** and **/media/nas6tb** is the directory the LXC container will use to access the NAS drives.

**Reply by Citroene** [reply]
Thx! this works for me!

**Reply by ameran** [reply]
Hi, 

I have a similar problem.
I've mounted the NAS share and I can browse to it via shell from the container, but I can't seem to change it in the webui.
The mount point is set to mp0: /mnt/downloads,mp=/mnt/
I've set the default save path to /mnt/ in the webui and the downloads keep stalling.
What am I doing wrong?

**Reply by oOStroudyOo** [reply]
@ameran does `/mnt/downloads` match the Path showing in the Datacenter storage window?

Below is how mine looks, notice Proxmox auto set mine as /mnt/pve/"storage name"

![image](https://github.com/tteck/Proxmox/assets/111181956/d67aa1ed-01f4-484c-89fc-c417ea0806d0)

**Reply by ameran** [reply]
@oOStroudyOo I tried a few different ways, but gave up and canned the VM.
I changed my approach and went for doing it all within the TrueNAS Scale VM.
Thanks for the assistance anyway.
